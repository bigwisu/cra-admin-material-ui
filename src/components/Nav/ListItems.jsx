import React from 'react';
import { Link } from 'react-router-dom';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListSubheader from '@material-ui/core/ListSubheader';
import DashboardIcon from '@material-ui/icons/Dashboard';
import PeopleIcon from '@material-ui/icons/People';
import BarChartIcon from '@material-ui/icons/BarChart';
import LayersIcon from '@material-ui/icons/Layers';
import AssignmentIcon from '@material-ui/icons/Assignment';
import AccessibilityNew from '@material-ui/icons/AccessibilityNew';
import SentimentSatisfiedAlt from '@material-ui/icons/SentimentSatisfiedAlt';
import ViewColumn from '@material-ui/icons/ViewColumn';

export const mainListItems = (
  <div>
    <ListItem button component={Link} to="/">
      <ListItemIcon>
        <DashboardIcon />
      </ListItemIcon>
      <ListItemText primary="Dashboard" />
    </ListItem>
    <ListItem button component={Link} to="/recipient">
      <ListItemIcon>
        <AccessibilityNew />
      </ListItemIcon>
      <ListItemText primary="Recipient" />
    </ListItem>
    <ListItem button component={Link} to="/donor">
      <ListItemIcon>
        <SentimentSatisfiedAlt />
      </ListItemIcon>
      <ListItemText primary="Donor" />
    </ListItem>
    <ListItem button>
      <ListItemIcon>
        <LayersIcon />
      </ListItemIcon>
      <ListItemText primary="Area" />
    </ListItem>
    <ListItem button>
      <ListItemIcon>
        <ViewColumn />
      </ListItemIcon>
      <ListItemText primary="Activity" />
    </ListItem>
    <ListItem button>
      <ListItemIcon>
        <BarChartIcon />
      </ListItemIcon>
      <ListItemText primary="Report" />
    </ListItem>
  </div>
);

export const adminListItems = (
  <div>
    <ListSubheader inset>Admin</ListSubheader>
    <ListItem button component={Link} to="/users">
      <ListItemIcon>
        <PeopleIcon />
      </ListItemIcon>
      <ListItemText primary="Users" />
    </ListItem>
    <ListItem button>
      <ListItemIcon>
        <AssignmentIcon />
      </ListItemIcon>
      <ListItemText primary="Last quarter" />
    </ListItem>
  </div>
);
