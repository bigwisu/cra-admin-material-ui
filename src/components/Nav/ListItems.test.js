import React from 'react';
import renderer from 'react-test-renderer';
import { mainListItems, adminListItems } from './ListItems';

describe('MainListItems snapshots', () => {
  test('MainListItems snapshot test', () => {
    const component = renderer.create(<mainListItems />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('adminListItems snapshots', () => {
  test('adminListItems snapshot test', () => {
    const component = renderer.create(<adminListItems />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
