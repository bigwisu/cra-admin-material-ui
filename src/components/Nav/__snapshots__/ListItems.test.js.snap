// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainListItems snapshots MainListItems snapshot test 1`] = `<mainListItems />`;

exports[`adminListItems snapshots adminListItems snapshot test 1`] = `<adminListItems />`;
